runtime/runtime/src/lib.rs

```diff
@@ -3273,6 +3246,44 @@ mod tests {
         assert_eq!(root_before, root_after, "state root changed for applying empty receipts");
     }
 
+    /// Test that delayed receipts are accounted for in the congestion info of
+    /// the ApplyResult.
+    #[test]
+    fn test_congestion_delayed_receipts_accounting() {
+        let initial_balance = to_yocto(10);
+        let initial_locked = to_yocto(0);
+        let deposit = to_yocto(1);
+        let gas_limit = 1;
+        let (runtime, tries, root, apply_state, _, epoch_info_provider) =
+            setup_runtime(initial_balance, initial_locked, gas_limit);
+
+        let n = 10;
+        let receipts = generate_receipts(deposit, n);
+
+        let apply_result = runtime
+            .apply(
+                tries.get_trie_for_shard(ShardUId::single_shard(), root),
+                &None,
+                &apply_state,
+                &receipts,
+                &[],
+                &epoch_info_provider,
+                Default::default(),
+            )
+            .unwrap();
+
+        assert_eq!(n - 1, apply_result.delayed_receipts_count);
```

Comment:
Just for my information, why is only one delayed receipt processed?

Reply:
Do you mean technically, why the code works that way? Then the answer is because of `let gas_limit = 1;` a few lines above.
Or if you wan to know why the test doesn't drain all receipts in a loop, then the answer is just to keep the test somewhat simple.

Reply:
I meant the former, thanks for explaining that makes sense.